lib/cfnguardian/tagger.rb

```diff
@@ -0,0 +1,71 @@
+require 'aws-sdk-cloudwatch'
+require 'cfnguardian/cloudwatch'
+require 'cfnguardian/log'
+
+module CfnGuardian
+  class Tagger
+    include Logging
+
+    def initialize()
+      @client = Aws::CloudWatch::Client.new()
+    end
+
+    def tag_alarm(alarm, global_tags={})
+      alarm_arn = CfnGuardian::CloudWatch.get_alarm_arn(alarm)
+      
+      new_tags = get_tags(alarm, global_tags)
+      current_tags = get_alarm_tags(alarm_arn)
+
+      tags_to_delete = get_tags_to_delete(current_tags, new_tags)
+
+      if tags_to_delete.any?
+        logger.debug "Removing tags #{tags_to_delete} from alarm #{alarm_arn}"
+        @client.untag_resource({
+          resource_arn: alarm_arn,
+          tag_keys: tags_to_delete
+        })
+      end
+
+      if tags_changed?(current_tags, new_tags)
+        # Aws::CloudWatch::Errors::Throttling
```

Comment:
What's this comment here for?

Reply:
it was a reminder to handle throttling errors which is now applied